bookkeeper-server/src/test/java/org/apache/bookkeeper/test/BookKeeperClusterTestCase.java

```diff
@@ -558,26 +559,16 @@ public int startNewBookie()
     protected BookieServer startBookie(ServerConfiguration conf)
             throws Exception {
         BookieServer server = new BookieServer(conf);
-        server.start();
-
+        BookieSocketAddress address = Bookie.getBookieAddress(conf);
         if (bkc == null) {
             bkc = new BookKeeperTestClient(baseClientConf);
         }
+        Future<?> waitForBookie = bkc.waitForBookie(address);
 
-        int port = conf.getBookiePort();
-        String host = InetAddress.getLocalHost().getHostAddress();
-        if (conf.getUseHostNameAsBookieID()) {
-            host = InetAddress.getLocalHost().getCanonicalHostName();
-        }
-        
-        while (conf.isForceReadOnlyBookie()
```

Comment:
will change

Reply:
Changing the original behaviour because it looks incomplete. Now its, wait for bookie in read only if is force read only, otherwise wait for bookie in writable.